packages/stylist/src/style_src.rs

```diff
@@ -50,22 +50,47 @@ pub struct StyleSource<'a> {
     _marker: PhantomData<&'a ()>,
 
     manager: Option<StyleManager>,
+    #[cfg(debug_assertions)]
+    pub(crate) location: String,
 }
 
 impl StyleSource<'_> {
-    pub(crate) fn try_to_sheet(&self) -> Result<Sheet> {
+    #[cfg(debug_assertions)]
+    #[track_caller]
+    pub(crate) fn get_location() -> String {
+        let caller_loc = std::panic::Location::caller();
+        // follow css class name rules:
+        // -?[_a-zA-Z]+[_a-zA-Z0-9-]*
+        let mut approximate_file: String = caller_loc
+            .file()
+            .chars()
+            .map(|c| if c.is_ascii_alphabetic() { c } else { '-' })
+            .collect();
+        // max len of 32 character, mostly sufficient
+        let mut split = approximate_file.len().saturating_sub(32);
+        while approximate_file.as_bytes().get(split) == Some(&b'-') {
+            split += 1;
+        }
+        // at most 32 characters. Long enough?
+        let file = approximate_file.split_off(split);
```

Comment:
When the file name is short enough, this can leak the path beyond project dir.

Reply:
As far as I know, `Location::file` gives not guarantees about the output even begin a filename to begin with. I tested it locally, `file()` only contains the path relative to the workspace roots (for the example folders in stylist).

Reply:
In any case, if you know a way to get actual module names instead of file paths, I'd happily use that - e.g. `std::backtrace` is nightly-only and I didn't find an alternative.

Reply:
Bumped to 64 characters. Any longer and it's less readable in the output html, imo.

Reply:
How about `std::module_path` (passed to `StyleSource` by `css!` via a hidden method like `with_manager`)?

What the example above is saying is that if debugging information is implemented by Yew and VComp is accepting component type as a generic, which means type name is possible with `std::any::type_name` (or `std::module_path` in `html!`) and hence can be passed to the child elements of a Virtual DOM.

Reply:
Yep; didn't find that macro, but it makes sense in ones generated from `css!`. Doesn't work for parser generated ones, though, so I think using `#[track_caller]` there makes sense.

Reply:
Honestly, after trying it out, I think `file!()` offers better information than `module_path`, at least for debugging.